### ISSUE TITLE
Change from DiffEqBase to SciMLBase

### ISSUE THREAD

**Issue by ChrisRackauckas** [CliMA/ClimaCore.jl#331]
DiffEqBase is the lowest common denominator for the DiffEq packages, not necessarily the whole SciML ecosystem, and so it has a lot DiffEq dependencies. These are generally not required by downstream packages. If what you're looking for is a way to define problems without having most dependencies, we recommend you use SciMLBase as the dependency since everything like ODEProblem, SteadyStateProblem, etc. is defined there. We basically recommend depending on SciMLBase for problem definitions, and solver packages for specific solvers, but generally most non-SciML packages should not be depending on DiffEqBase directly (given the split of SciMLBase in 2021)

For more details see: https://diffeq.sciml.ai/stable/features/low_dep/ and https://discourse.julialang.org/t/psa-the-right-dependency-to-reduce-from-differentialequations-jl/72757

Let me know if you need any help updating this, though for almost all dependents here it should be a trivial name change as you're actually using pieces from SciMLBase.

**Comment by jakebolewski** [comment]
ArrayParition has been removed in lieu of consistently using FieldVectors, it would be good to go back and see how much of the shims we can remove now in the above linked file.  We might only need a couple overloads of SciML base in the end.

**Comment by jakebolewski** [comment]
@ChrisRackauckas it looks like we still need an overload to `DiffEqBase.calculate_residuals!` https://github.com/CliMA/ClimaCore.jl/blob/main/src/Fields/compat_diffeq.jl#L50

**Comment by jakebolewski** [comment]
The stracktrace for context is:
```
DiffEq Solvers: Error During Test at /home/jake/clima/ClimaCore.jl/test/fielddiffeq.jl:9
  Got exception outside of a @test
  ArgumentError: broadcasting over dictionaries and `NamedTuple`s is reserved
  Stacktrace:
    [1] broadcastable(#unused#::NamedTuple{(:c, :s), Tuple{Float64, Float64}})
      @ Base.Broadcast ./broadcast.jl:705
    [2] broadcasted
      @ ./broadcast.jl:1300 [inlined]
    [3] calculate_residuals
      @ ~/.julia/packages/DiffEqBase/b1nST/src/calculate_residuals.jl:23 [inlined]
    [4] _broadcast_getindex_evalf
      @ ./broadcast.jl:670 [inlined]
    [5] _broadcast_getindex
      @ ./broadcast.jl:643 [inlined]
    [6] getindex
      @ ./broadcast.jl:597 [inlined]
    [7] copyto!
      @ ~/clima/ClimaCore.jl/src/DataLayouts/broadcast.jl:361 [inlined]
    [8] copyto!(dest::IJFH{NamedTuple{(:c, :s), Tuple{Float64, Float64}}, 4, Array{Float64, 4}}, bc::Base.Broadcast.Broadcasted{ClimaCore.DataLayouts.IJFHStyle{4, Array{Float64}}, Nothing, typeof(DiffEqBase.calculate_residuals), Tuple{IJFH{NamedTuple{(:c, :s), Tuple{Float64, Float64}}, 4, Array{Float64, 4}}, IJFH{NamedTuple{(:c, :s), Tuple{Float64, Float64}}, 4, Array{Float64, 4}}, IJFH{NamedTuple{(:c, :s), Tuple{Float64, Float64}}, 4, Array{Float64, 4}}, Float64, Float64, Base.RefValue{typeof(DiffEqBase.ODE_DEFAULT_NORM)}, Float64}})
      @ ClimaCore.DataLayouts ~/clima/ClimaCore.jl/src/DataLayouts/broadcast.jl:336
    [9] copyto!
      @ ~/clima/ClimaCore.jl/src/Fields/broadcast.jl:97 [inlined]
   [10] materialize!
      @ ./broadcast.jl:871 [inlined]
   [11] materialize!
      @ ./broadcast.jl:868 [inlined]
   [12] fast_materialize!
      @ ~/.julia/packages/FastBroadcast/PYsFz/src/FastBroadcast.jl:31 [inlined]
   [13] calculate_residuals!
      @ ~/.julia/packages/DiffEqBase/b1nST/src/calculate_residuals.jl:94 [inlined]
   [14] perform_step!(integrator::OrdinaryDiffEq.ODEIntegrator{Tsit5{typeof(OrdinaryDiffEq.trivial_limiter!), typeof(OrdinaryDiffEq.trivial_limiter!), Static.False}, true, ClimaCore.Fields.Field...)
...
```

**Comment by charleskawczynski** [comment]
Closed by #1414
